service/k8sclient/kubectl/kubectl.go

```diff
@@ -122,27 +123,33 @@ func getKubectlCmd(ctx context.Context) ([]string, error) {
 		return nil, err
 	}
 
-	return lookupCorrectKubectlCmd(kubectlCmdNames)
+	return lookupCorrectKubectlCmd(defaultKubectl, kubectlCmdNames)
 }
 
-func lookupCorrectKubectlCmd(kubectlCmdNames []string) ([]string, error) {
+func lookupCorrectKubectlCmd(defaultKubectl, kubectlCmdNames []string) ([]string, error) {
 	for _, kubectlCmdName := range kubectlCmdNames {
-		kubectlPath, err := exec.LookPath(kubectlCmdName)
+		cmd := strings.Split(kubectlCmdName, " ")
+		kubectlPath, err := exec.LookPath(cmd[0])
 		if err == nil {
-			return []string{kubectlPath}, nil
+			return append([]string{kubectlPath}, cmd[1:]...), nil
 		}
 	}
 
-	// if none found (pass empty kubectlCmdNames) use default version of kubectl.
-	return strings.Split(defaultKubectl, " "), nil
+	if defaultKubectl == nil {
+		return nil, errors.Errorf("cannot find kubectl: %v, %v", defaultKubectl, kubectlCmdNames)
+	}
+
+	// if none found and default is not empty use default version of kubectl.
+	return defaultKubectl, nil
```

Comment:
What's is the reason to use `defaultKubectl`? if we got kubernetes cluster with version 1.19, but we have only kubectl-1.16 can we be sure that everything will work correctly? And should we try?

Reply:
@AlekSi what do you think?

Reply:
We should have _some_ fallback if we could not pick better version

Reply:
ok